src/App.jsx

```diff
@@ -14,45 +16,124 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function getPrepearedGoods(goods, { currentSortField, reversedState }) {
+  const prepearedGoods = [...goods];
+
+  if (currentSortField === '') {
+    if (reversedState === 'reversed') {
+      prepearedGoods.reverse();
+    }
+  }
+
+  if (currentSortField === 'Name') {
+    if (reversedState === 'reversed') {
+      prepearedGoods.sort().reverse();
+    }
+
+    if (reversedState !== 'reversed') {
+      prepearedGoods.sort();
+    }
+  }
+
+  if (currentSortField === 'Length') {
+    if (reversedState === 'reversed') {
+      prepearedGoods.sort(
+        (good1, good2) => (good1.length - good2.length),
+      ).reverse();
+    }
+
+    if (reversedState !== 'reversed') {
+      prepearedGoods.sort((good1, good2) => (good1.length - good2.length));
+    }
+  }
+
+  return prepearedGoods;
```

Comment:
i think you can do this code simplier and shorter. Try just add  prepearedGoods.reverse() in the end, after sort by length and sort, without if (reversedState === 'reversed') for sort